registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/UserNotificationApi.java

```diff
@@ -0,0 +1,316 @@
+
+	@Value("${vertx.cluster.configuration}")
+	private String clusterManagerUrl;
+
+	/** server port number. */
+	@Value("${mosip.regproc.user.notification.api.server.port}")
+	private String port;
+
+	@Autowired
+	UserNotificationApi UserNotificationApi;
+	
+	/** The web sub util. */
+	@Autowired
+	private WebSubUtil webSubUtil;
+
+	/** worker pool size. */
+	@Value("${worker.pool.size}")
+	private Integer workerPoolSize;
+
+	@Value("${mosip.regproc.user.notification.api.eventbus.port}")
+	private String eventBusPort;
+
+	@Value("${mosip.registration.processor.datetime.pattern}")
+	private String dateTimePattern;
+
+	@Value("${mosip.regproc.user.notification.api.id}")
+	private String id;
+
+	@Value("${mosip.regproc.user.notification.api.version}")
+	private String version;
+	
+	private static final String STAGE_PROPERTY_PREFIX = "mosip.regproc.reprocessor.";
+
+	@Autowired
+	MosipRouter router;
+
+	@Value("${server.servlet.path}")
+	private String contextPath;
+
+	private MosipEventBus mosipEventBus = null;
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(UserNotificationApi.class);
+
+	/** The registration status service. */
+	@Autowired
+	RegistrationStatusService<String, InternalRegistrationStatusDto, RegistrationStatusDto> registrationStatusService;
+
+	/** The module id. */
+	public static String MODULE_ID = PlatformSuccessMessages.RPR_USER_NOTIFICATION_API_SUCCESS.getCode();
+
+	/** The module name. */
+	public static String MODULE_NAME = ModuleName.USER_NOTIFICATION_API.toString();
+
+	/** The core audit request builder. */
+	@Autowired
+	AuditLogRequestBuilder auditLogRequestBuilder;
+
+	@Autowired
+	UserNotificationRequestValidator UserNotificationRequestValidator;
+
+	/**
+	 * Deploy verticle.
+	 */
+	public void deployVerticle() {
+		mosipEventBus = this.getEventBus(this, clusterManagerUrl, workerPoolSize);
+	}
+
+	@Override
+	public void start() {
+		router.setRoute(this.postUrl(getVertx(), null, null));
+		this.routes(router);
+		this.createServer(router.getRouter(), Integer.parseInt(port));
+	}
+
+	@Override
+	public Integer getEventBusPort() {
+		return Integer.parseInt(eventBusPort);
+	}
+
+	/**
+	 * contains all the routes in this stage
+	 *
+	 * @param router
+	 */
+	private void routes(MosipRouter router) {
+		router.post(contextPath + "/userNotification");
+		router.handler(this::processURL, this::failure);
+	}
+
+	/**
+	 * method to process the context received.
+	 *
+	 * @param ctx the ctx
+	 */
+	public void processURL(RoutingContext ctx) {
+		LogDescription description = new LogDescription();
+		InternalRegistrationStatusDto registrationStatusDto;
+		JsonObject obj = ctx.getBodyAsJson();
+		String rid = null;
+		try {
+			UserNotificationDTO UserNotificationDTO = JsonUtil.readValueWithUnknownProperties(obj.toString(),
+					UserNotificationDTO.class);
+			UserNotificationRequestDTO UserNotificationRequestDTO = UserNotificationDTO.getRequest();
```

Comment:
"UserNotificationRequestDTO" and "UserNotificationDTO" - camel case not followed

Reply:
Done